plugins/modules/keycloak_realm_rolemapping.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) Ansible project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_realm_rolemapping
+
+short_description: Allows administration of Keycloak realm role mappings into groups with the Keycloak API
+
+version_added: 8.3.0
+
+description:
+    - This module allows you to add, remove or modify Keycloak realm role
+      mappings into groups with the Keycloak REST API. It requires access to the
+      REST API via OpenID Connect; the user connecting and the client being used
+      must have the requisite access rights. In a default Keycloak installation,
+      admin-cli and an admin user would work, as would a separate client
+      definition with the scope tailored to your needs and a user having the
+      expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/18.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating a group_rolemapping, where possible provide the role ID to the module. This removes a lookup
+      to the API to translate the name into the role ID.
+
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: full
+
+options:
+    state:
+        description:
+            - State of the realm_rolemapping.
+            - On C(present), the realm_rolemapping will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the realm_rolemapping will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this role_representation resides.
```

Comment:
Here and in a few more places role_representation shows up. Should that be realm_rolemapping?

Reply:
I don't think so. This module is about mapping a realm's roles (i.e. their role_representation) to a group. The wording is taken from [`keycloak_client_rolemapping`](https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_client_rolemapping_module.html#parameter-realm)